Feat/dongfa/BottomNavigationBar

dong-fa opened the pull request:
Icon.tsx 오타 수정 밎 누락된 svg 추가
BottomNavi 타입 정의

dong-fa committed:
Feat #16 BottomNavigation 제작

Icon.tsx 오타 수정 밎 누락된 svg 추가
BottomNavi 타입 정의

dong-fa committed:
Desigh #16 BotNavi 임의 CSS 수정

jennie-0530 commented on src/components/navigation/BottomNavigationBar.tsx:
```diff
@@ -0,0 +1,99 @@
+  width: 1024px;
+  height: 47px;
+  background: ${color.basicWhite};
+
```
엔터 멈춰!

dong-fa replied:
엔터 정리 할게영

0SCAR0421 commented on src/components/navigation/BottomNavigationBar.tsx:
```diff
@@ -0,0 +1,99 @@
+import { Caption2 } from '../typography';
+import { IBottomNavigationBarProps } from '../../types/BottomNavigationBarTypes';
+
+const getPath = () => {
+  const { pathname } = new URL(window.location.href);
+
+  return pathname.slice(1);
+};
```
이 함수는 원하는 값이 도출안될수도 있을거 같아요

ex) detail/123 이면 원하는 값이 detail인데 detail/123으로 도출될거 같습니다.

0SCAR0421 replied:
이친구는 util함수가 아닐까요?

dong-fa replied:
해당 함수 util로 빼고 path값 다시 확인 해보겠습니다.

sungin46 reviewed (commented):
깔쌈한 코드네요 🙏

dong-fa committed:
Feat #16 BotNaviBar 피드백 반영

엔터 지옥 정리

Util 생성
Util 정의: 함수에서 Path가 깊어지는 경우 pathname을 쪼개서 botNaviBar 랜더링에 필요한 string만 사용

사용법은 readme에 따로 정리 예정

dong-fa committed:
fix #16 console.log 삭제

sungin46 commented on src/util/index.ts:
```diff
@@ -1,4 +1,4 @@
 export * from './buttonUtils';
-export * from './itemUtils';
```
itemUtils 익스포트 빠진거같아오

dong-fa replied:
base 브랜치가 Item 컴포넌트 풀되기전이라 삭제했습니다.

jennie-0530 commented on src/types/BottomNavigationBarTypes.ts:
```diff
@@ -0,0 +1,14 @@
+    center: string;
+    mypage: string;
+  };
+  onHomeClick: () => void;
```
저희 코드 컨벤션 정할 때 `on + 이벤트명 + 동작대상` 으로 하지 않았나용?

dong-fa replied:
컨벤션 적용해서 커밋 할게용

dong-fa committed:
FIx #16 피드백 2차 반영

이벤트 함수 컨벤션 적용

jennie-0530 commented on src/components/navigation/BottomNavigationBar.tsx:
```diff
@@ -0,0 +1,91 @@
+const BottomNavigationBar = ({
+  tabNameLists,
+  onClickHome,
+  onClickReserve,
```
`onClickReservation`이 더 명확한 것 같아용

jennie-0530 replied:
Partiend는 무슨 뜻인지 와닿지가 않읍니다. `onClickMembership` 은 어떠신가요?

dong-fa replied:
오타 수정 했습니다.

dong-fa committed:
FIx #16 피드백 반영 3차

Icon 오타 수정
reserve => reservation으로 수정
BottomNavigationTypes => bottomNavigationTypes

dong-fa committed:
FIx #16 피드백 반영 3차

Icon 오타 수정
reserve => reservation으로 수정
BottomNavigationTypes => bottomNavigationTypes

dong-fa commented:
App.tsx 체크 푼줄 알았는데 다시 걸려서 올라갔네요 수정해서 머지 요청합니다.

sungin46 committed:
Merge branch 'dev' into Feat/dongfa/BotNavibar
